File: src/core/configuration.cpp
#include "configuration.h"

namespace mavsdk {

Configuration::Configuration(uint8_t system_id, uint8_t component_id) :
    _address{system_id, component_id},
    _usage_type(Configuration::UsageType::Custom) {
}

Configuration::Configuration(UsageType usage_type) :
    _address{245, 1},
    _usage_type(usage_type) {
}

Configuration::~Configuration() {}

uint8_t Configuration::get_system_id() const {
    return _address.system_id;
}

uint8_t Configuration::get_component_id() const {
    return _address.component_id;
}

Configuration::UsageType Configuration::get_usage_type() const {
    return _usage_type;
}

} // namespace mavsdk

File: kt/mavsdk-kotlin/src/jvmAndroidMain/kotlin/io/mavsdk/kotlin/Configuration.kt
package io.mavsdk.kotlin

import io.mavsdk.jni.NativeConfiguration

actual class Configuration private constructor(private val handle: Long) : AutoCloseable {

    actual var systemId: Int
        get() = NativeConfiguration.getSystemId(handle)
        set(value) = NativeConfiguration.setSystemId(handle, value)

    actual var componentId: Int
        get() = NativeConfiguration.getComponentId(handle)
        set(value) = NativeConfiguration.setComponentId(handle, value)

    actual override fun close() {
        NativeConfiguration.destroy(handle)
    }

    actual internal fun getHandle(): Long = handle

    actual companion object {
        init {
            NativeLibraryLoader.loadLibrary()
        }

        actual fun createWithComponentType(componentType: ComponentType): Configuration =
            Configuration(NativeConfiguration.createWithComponentType(componentType.value))

        actual fun createManual(
            systemId: Int,
            componentId: Int,
            alwaysSendHeartbeats: Boolean,
        ): Configuration =
            Configuration(
                NativeConfiguration.createManual(systemId, componentId, alwaysSendHeartbeats)
            )
    }
}
